ledger/eval/eval.go

```diff
@@ -790,40 +791,30 @@ func StartEvaluator(l LedgerForEvaluator, hdr bookkeeping.BlockHeader, evalOpts
 		return nil, fmt.Errorf("overflowed subtracting rewards for block %v", hdr.Round)
 	}
 
-	eligible, err := eval.eligibleForIncentives(prevHeader.Proposer)
-	if err != nil {
-		return nil, err
-	}
-
-	if eligible {
-		incentive, _ := basics.NewPercent(proto.Mining().Percent).DivvyAlgos(prevHeader.FeesCollected)
-		total, o := basics.OAddA(incentive, prevHeader.Bonus)
-		if o {
-			return nil, fmt.Errorf("overflowed adding bonus incentive %d %d", incentive, prevHeader.Bonus)
-		}
-
-		sink, err := eval.state.lookup(prevHeader.FeeSink)
-		if err != nil {
-			return nil, err
-		}
-		available := sink.AvailableBalance(&proto)
-		total = basics.MinA(total, available)
-
-		err = eval.state.Move(prevHeader.FeeSink, prevHeader.Proposer, total, nil, nil)
-		if err != nil {
-			// Should be impossible, we used AvailableBalance
-			return nil, fmt.Errorf("unable to pay block incentive: %v", err)
+	// Move last block's proposer payout to the proposer
+	if !prevHeader.Proposer.IsZero() {
+		// Payout the previous proposer.
+		if !prevHeader.ProposerPayout.IsZero() {
+			// Use the FeeSink from that header, since that's the checked
+			// balance. (Though we have no expectation it will ever change.)
+			err = eval.state.Move(prevHeader.FeeSink, prevHeader.Proposer, prevHeader.ProposerPayout, nil, nil)
+			if err != nil {
+				// Should be impossible, it was checked when put into the block
+				return nil, fmt.Errorf("unable to payout block incentive: %v", err)
+			}
 		}
-	}
 
-	// Note their proposal if MiningEnabled.
-	if proto.Mining().Enabled {
+		// Increment LastProposed on the proposer account
 		prp, err := eval.state.Get(prevHeader.Proposer, false)
 		if err != nil {
 			return nil, err
 		}
 		prp.LastProposed = hdr.Round - 1
-		// An account could propose, even while suspended, because of the 320 round lookback.
+
+		// An account could propose, even while suspended, because of the 320
+		// round lookback.  Doing so is evidence the account is
+		// back. Unsuspend. But the account will remain not IncentiveElgible
+		// until they repay the initial fee.
```

Comment:
Usability-wise, how will they know they need to pay the inital fee if that is the thing stopping them?